File: ios/stereoCamera/stereoCamera/Comm/Commands/SetFacing.swift
//
//  SetCamera.swift
//  stereoCamera
//
//  Created by hallmarklabs on 6/27/18.
//  Copyright © 2018 cody. All rights reserved.
//

import Foundation

class SetFacing : Command
{
    var facing:Bool = false

    override init()
    {
        super.init()
        doInit()
    }
    
    init(_ facing:Bool)
    {
        super.init()
        self.facing = facing
        doInit()
    }
    
    private func doInit()
    {
        cmdtype = CommandTypes.SET_FACING
        expectsResponse = true
    }
    
    override func send(comm: Comm) -> Bool
    {
        let result = super.send(comm: comm)
        if (!result) { return false }
        
        var bytes:[UInt8] = []
        bytes += Bytes.toByteArray(facing)
        
        let sz = comm.write(buf: bytes)
        return (sz <= 0) ? false : true
    }
    
    override func receive(comm: Comm) -> Bool
    {
        let result = super.receive(comm: comm)
        if (!result) { return false }
        
        let (buf, sz) = comm.read(sz: 1)
        if (sz <= 0) { return false }
        
        facing = Bytes.fromByteArray(buf)
        return true
    }
}

File: android/app/src/main/java/com/munger/stereocamera/ip/command/commands/SetFacing.java
package com.munger.stereocamera.ip.command.commands;

import com.munger.stereocamera.ip.command.Comm;
import com.munger.stereocamera.ip.command.Command;

import java.io.IOException;

public class SetFacing extends Command
{
	public boolean facing = false;

	public SetFacing()
	{
		super();
		doInit();
	}

	public SetFacing(boolean facing)
	{
		super();
		this.facing = facing;
		doInit();
	}

	private void doInit()
	{
		cmdtype = Type.SET_FACING;
		expectsResponse = true;
	}

	@Override
	public boolean send(Comm comm)
	{
		boolean result = super.send(comm);
		if (!result)
			{ return false; }

		try
		{
			comm.putByte((facing) ? (byte) 1 : (byte) 0);
		}
		catch(IOException e){
			return false;
		}

		return true;
	}

	@Override
	public boolean receive(Comm comm)
	{
		boolean result = super.receive(comm);
		if (!result)
			{ return false; }

		try
		{
			facing = comm.getBoolean();
		}
		catch(IOException e){
			return false;
		}

		return true;
	}
}
